src/pipeline.rs

```diff
@@ -572,10 +574,15 @@ pub(crate) async fn pipeline_task(
             }
             AnyMsg::Mgmt(MgmtMsg::ConnectInput {
                 endpoint,
+                port,
                 target,
                 is_transactional,
             }) => {
-                info!("{ctx} Connecting {endpoint} to port 'in'");
+                info!("{ctx} Connecting '{endpoint}' to port '{port}'");
+                if !input_does_exist(&port, &pipeline) {
+                    error!("{ctx} Error connecting input pipeline '{port}' as it does not exist",);
```

Comment:
If we add a sender to the `ConnectInput` for sending back a result, like we do in `Inspect` below, we could completely error the startup and not just emit a log message. The code sending that message would need to create a channel and wait for a response then.

Reply:
Thank you for the suggestion Mattias! I have added a transmitter and receiver in `ConnectInput`